app/models/sync_log.rb

```diff
@@ -7,11 +7,27 @@ class SyncLog < ApplicationRecord
   validates :started_at, presence: true
   validates :user, presence: true
 
-  def self.start(task_name:, user:)
-    create!(task_name:, user:, started_at: Time.current)
+  def self.start!(task_name:)
+    raise StandardError, 'cannot start a task when one is already in progress' if latest(task_name)&.in_progress?
```

Comment:
Will need user here for the if condition too

Reply:
I think this one is OK since we share the same scope as `start!`?